packages/react-components/src/components/xy-output-component.tsx

```diff
@@ -564,37 +578,90 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
         }
     }
 
-    private tooltip(x: number, y: number): void {
+    private closestPoint(array: any, point: any, xRatio: number, yRatio: number, chartHeight: number): number | undefined {
+        const hypos: any[] = [];
+
+        array.forEach((ar: any) => {
+            // Convert x values to screen pixels
+            const pointX = point.x * xRatio;
+            const x = ar.x * xRatio;
+
+            // Convert y values to screen pixels
+            const pointY: number = chartHeight - point.y;
+            const y = (ar.y - this.state.allMin) * yRatio;
+
+            // Calculate hypotenuses
+            const hypotenuse = Math.sqrt(Math.pow(Math.abs(pointX - x), 2) + Math.pow(Math.abs(pointY - y), 2));
+            hypos.push(hypotenuse);
+        });
+
+        const min = Math.min(...hypos);
+
+        if (min < 20) {
+            const index = hypos.indexOf(min);
+            return index;
+        }
+        else {
+            return undefined;
+        }
+    }
+
+    private tooltip(x: number, y: number, bounds: DOMRect | undefined): void {
         const xPos = this.positionXMove;
         const timeForX = this.getTimeForX(xPos);
         let timeLabel: string | undefined = timeForX.toString();
         if (this.props.unitController.numberTranslator) {
             timeLabel = this.props.unitController.numberTranslator(timeForX);
         }
         const chartWidth = this.chartRef.current.chartInstance.width;
+        const chartHeight = this.chartRef.current.chartInstance.height;
         const arraySize = this.state.xyData.labels.length;
         const index = Math.max(Math.round((xPos / chartWidth) * (arraySize - 1)), 0);
+        const offset = this.props.viewRange?.getOffset() ?? BigInt(0);
+        const end = Number(this.props.viewRange.getEnd() - offset);
+        const xRatio = chartWidth / end;
```

Comment:
This should probably be chartWidth / (end - start). But there is still something wrong...

Reply:
Using `end - start` instead of `end - offset` fixed it.